apps/users/dataloaders.py

```diff
@@ -39,6 +45,20 @@ def batch_load_fn(self, keys):
                 'portfolio_role_display': portfolio_role_display,
             }
 
-        return Promise.resolve([
-            _map[key] for key in keys
-        ])
+        return Promise.resolve([_map.get(key, default_guest_data()) for key in keys])
+
+
+def default_guest_data():
+    """
+    Provides default guest data for users without specific roles.
+
+    Returns:
+        dict: A dictionary representing the default guest user portfolio metadata.
+    """
+    return {
+        'is_admin': False,
+        'is_directors_office': False,
+        'is_reporting_team': False,
+        'portfolio_role': USER_ROLE.GUEST.value,
+        'portfolio_role_display': USER_ROLE.GUEST.label,
+    }
```

Comment:
For now let's define this locally as it might not be usable for other places due to specific data structure.

```suggestion
        return Promise.resolve([
           # For default, using GUEST role
            _map.get(key, {
                'is_admin': False,
                'is_directors_office': False,
                'is_reporting_team': False,
                'portfolio_role': USER_ROLE.GUEST.value,
                'portfolio_role_display': USER_ROLE.GUEST.label,
            })
            for key in keys
        ])
```